telemetry/csharp/AwsToolkit.Telemetry.Events.Generator/DefinitionsBuilder.cs

```diff
@@ -21,6 +21,13 @@ public class DefinitionsBuilder
         private const string AddMetadataMethodName = "AddMetadata";
         private const string InvokeTransformMethodName = "InvokeTransform";
 
+        // contains metadata fields that should be skipped when generating code.
+        // These fields are covered by the class BaseTelemetryEvent
+        private static readonly string[] ImplicitFields = 
+        {
+            "reason",
+        };
```

Comment:
I guess reflection isn't too ergonomic in C#?

Reply:
I opted for the simplest/fastest route at this time. We could annotate the BaseTelemetryEvent class and then use reflection to determine these fields, but for now we have just the one.

Reply:
Oh you have to annotate it? Yeah that's not ergonomic.